### ISSUE TITLE
registerComponent basePath vs isBower

### ISSUE THREAD

**Issue by moyesm** [Allov/rc.knockout.utilities#1]
There's an issue here, and a question/discussion:

The issue is that passing the isBower param to registerComponent (or registerModal) clobbers basePath.

This becomes an issue because of the convention we're using when naming koco repos/files. In most cases, the component name doesn't match the file name. Ex: `koco-url-utilities/src/url-utilities`
If we set `isBower`, any basePath passed in gets replaced by `bower_components/url-utilities/src/url-utilities` (ie, with no `koco-`)

A user who misunderstands the purpose of isBower might get confused.

This also affects modals. 

```
    modaler.registerModal('confirm', {
                backdrop: 'static',
                keyboard: false,
                isBower: true,
                basePath: 'bower_components/koco-confirm-modal/src'
    });
```

As a first-time user, I might expect that I could use isBower _instead_ of basePath, but that won't work with any of our components. So, question: is isBower useful enough as-is? (Or, non-confusing enough?)
We could add a bowerPackageName param, which would accomplish the same thing. Is that... better?

**Comment by fcharest** [comment]
I agree, `isBower` may be confusing and the way basePath has been implemented is a bug, I think.

So, I spoke with @RCMax and we concluded that the `bowerPackageName` should be clear enough provided `basePath` would superseed all parameters.

In (pseudo) code:

``` javascript
if (basePath) {
  componentPath = basePath + "/" + componentName;
} else if (bowerPackageName) {
  componentPath = "bower_components/" + bowerPackageName + "/src/" + componentName;
} else {
  componentPath = "components/" + componentName + "/" + componentName;
}
```

Does that make sens?

**Comment by moyesm** [comment]
Yup, makes sense to me.
